src/Aspire.Hosting.Azure/Bicep/AzureBicepServiceBusResource.cs

```diff
@@ -39,20 +101,94 @@ public static class AzureBicepServiceBusExtensions
+    }
+
+    /// <summary>
+    /// Adds a Azure Service Bus Queue resource to the application model. This resource requires an <see cref="AzureServiceBusResource"/> to be added to the application model.
+    /// </summary>
+    /// <param name="serviceBusBuilder">The Azure Service Bus resource builder.</param>
+    /// <param name="name">The name of the queue.</param>
+    public static IResourceBuilder<AzureBicepServiceBusQueueResource> AddQueue(this IResourceBuilder<IResourceWithParent<AzureBicepServiceBusResource>> serviceBusBuilder, string name)
+    {
+        return serviceBusBuilder.ApplicationBuilder.AddQueue(name, serviceBusBuilder.Resource.Parent);
+    }
+
+    private static IResourceBuilder<AzureBicepServiceBusQueueResource> AddQueue(this IDistributedApplicationBuilder builder, string name, AzureBicepServiceBusResource parent)
+    {
+        var resource = new AzureBicepServiceBusQueueResource(name, parent);
+
+        parent.Queues.Add(name);
+
+        // TODO: This should not be needed here. It should be lazily evaluated.
+        parent.Parameters["queues"] = parent.Queues;
+
+        return builder.AddResource(resource)
                       .WithManifestPublishingCallback(resource.WriteToManifest);
     }
+
+    /// <summary>
+    /// Adds a Azure Service Bus Topic resource to the application model. This resource requires an <see cref="AzureServiceBusResource"/> to be added to the application model.
+    /// </summary>
+    /// <param name="serviceBusBuilder">The Azure Service Bus resource builder.</param>
+    /// <param name="name">The name of the topic.</param>
+    /// <param name="subscriptions">The name of the subscriptions.</param>
+    public static IResourceBuilder<AzureBicepServiceBusTopicResource> AddTopic(this IResourceBuilder<AzureBicepServiceBusResource> serviceBusBuilder, string name, string[] subscriptions)
+    {
+        return serviceBusBuilder.ApplicationBuilder.AddTopic(name, subscriptions, serviceBusBuilder.Resource);
+    }
+
+    /// <summary>
+    /// Adds a Azure Service Bus Topic resource to the application model. This resource requires an <see cref="AzureBicepServiceBusResource"/> to be added to the application model.
+    /// </summary>
+    /// <param name="serviceBusBuilder">The Azure Service Bus resource builder.</param>
+    /// <param name="name">The name of the topic.</param>
+    /// <param name="subscriptions">The name of the subscriptions.</param>
+    public static IResourceBuilder<AzureBicepServiceBusTopicResource> AddTopic(this IResourceBuilder<IResourceWithParent<AzureBicepServiceBusResource>> serviceBusBuilder, string name, string[] subscriptions)
+    {
+        return serviceBusBuilder.ApplicationBuilder.AddTopic(name, subscriptions, serviceBusBuilder.Resource.Parent);
+    }
+
+    private static IResourceBuilder<AzureBicepServiceBusTopicResource> AddTopic(this IDistributedApplicationBuilder builder, string name, string[] subscriptions, AzureBicepServiceBusResource parent)
```

Comment:
Should we just return `IResourceBuilder<AzureBicepServiceBusResource>`

Reply:
Yep